compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -682,12 +682,11 @@ void addMLIRAIRLoweringPasses(OpPassManager &passManager, AMDAIEDevice device,
   passManager.addPass(createCSEPass());
   {
     xilinx::air::AIRFuseChannelsOptions options;
-    std::vector<std::string> mode;
+    // static because clAggressiveMode is an ArrayRef
     if (useTilePipeline == TilePassPipeline::PackPeelPipeline &&
         matmulElementwiseFusion) {
-      mode.push_back("L1");
+      options.clAggressiveMode = {"L1"};
```

Comment:
Still UB me thinks. This is like 

ArrayRef<std::string> x = {"L1"}; 

which can't be good, where does the string actually live? 

But feel free to land because I don't know if caching works nicely for IREE bumps, and I promise to clean this later.

Reply:
I don't understand why you think is undefined behavior? What about this do you think is compiler/impl dependent? 

> where does the string actually live?

It's a local so I'm almost 100% it's stack allocation.  Regardless, the ctor I'm using here moves the string:

```
    /// The declaration here is extra complicated so that "arrayRef = {}"
    /// continues to select the move assignment operator.
```

If you're very worried I can link a godbolt to double-check.

Reply:
> List-initializing std::initializer_list
> The backing array has the same lifetime as any other temporary object, except that initializing an std::initializer_list object from the backing array extends the lifetime of the array exactly like binding a reference to a temporary.

https://en.cppreference.com/w/cpp/language/list_initialization

Reply:
I am still very worried. 

Here's what happens I think: 
1) You create your local ` xilinx::air::AIRFuseChannelsOptions options;` which is basically an ArrayRef<std::string>.
2) You pass your local `options` into `xilinx::air::createAIRFuseChannels(options)`. 
3) Inside that pass creator function, the ArrayRef is copied. This is just a copy of 2 stack pointers, the string is not copied!
4) So inside your pass object, you end storing pointers to stack. 
5) By the time you run the pass, that stack pointer is probably long since overwritten.

Reply:
Ok I concede I've been bamboozled by whoever implemented that ctor:

> Anyone using ArrayRef should already be aware of the fact that it does not do lifetime extension

https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/ADT/ArrayRef.h#L111

because in my understanding avoiding the kind of UB you're worried about is exactly what `std::initializer_list` was invented for. Ok then the solution is a `static llvm::SmallVector` because there's no other way for the reference to outlive the pass (since it's the passman that's passed in in this function). I'll patch it up before I land.

Reply:
Thanks. Yes I think using `static` is the only easy solution.